ingenialink/canopen/dictionary.py

```diff
@@ -92,162 +20,52 @@ class CanopenDictionary(Dictionary):
-            elif dtype == "u32":
-                dtype = REG_DTYPE.U32
-            elif dtype == "str":
-                dtype = REG_DTYPE.STR
-            else:
-                raise_err(lib.IL_EINVAL, 'Invalid data type')
-
-            # Access
-            access = register.attrib['access']
-            if access == "r":
-                access = REG_ACCESS.RO
-            elif access == "w":
-                access = REG_ACCESS.WO
-            elif access == "rw":
-                access = REG_ACCESS.RW
-            else:
-                raise_err(lib.IL_EACCESS, 'Invalid access type')
-
-            # Subnode
-            subnode = int(
-                register.attrib['subnode']) if 'subnode' in register.attrib else 1
-
-            # Storage
-            storage = register.attrib[
-                'storage'] if 'storage' in register.attrib else None
-            cat_id = register.attrib['cat_id'] if 'cat_id' in register.attrib else None
-            if 'internal_use' in register.attrib:
-                internal_use = register.attrib['internal_use']
-            else:
-                internal_use = 0
-
-            # Labels
-            labels_elem = register.findall('./Labels/Label')
-            labels = {label.attrib['lang']: label.text for label in labels_elem}
-
-            # Range
-            range_elem = register.find('./Range')
-            reg_range = (None, None)
-            if range_elem is not None:
-                range_min = range_elem.attrib['min']
-                range_max = range_elem.attrib['max']
-                reg_range = (range_min, range_max)
-
-            # Enumerations
-            enums_elem = register.findall('./Enumerations/Enum')
-            enums = [{enum.attrib['value']: enum.text} for enum in enums_elem]
-
-            reg = CanopenRegister(identifier, units, cyclic, idx, subidx, dtype,
-                                  access, subnode=subnode,
-                                  storage=storage, reg_range=reg_range,
-                                  labels=labels, enums=enums,
-                                  enums_count=len(enums), cat_id=cat_id,
-                                  internal_use=internal_use)
-            self.__registers[int(subnode)][identifier] = reg
-        except Exception as e:
-            pass
-
-    def registers(self, subnode):
-        """Gets the register dictionary to the targeted subnode.
-
-        Args:
-            subnode (int): Identifier for the subnode.
-
-        Returns:
-            dict: Dictionary of all the registers for a subnode.
-
-        """
-        return self.__registers[subnode]
+            current_read_register = super()._read_register(register)
+
+            current_read_register[AttrRegCanDict.IDX] = int(register.attrib['address'][:6], 16)
+            current_read_register[AttrRegCanDict.SUBIDX] = int("0x" + register.attrib['address'][-2:], 16)
+
+            return current_read_register
+
+        except KeyError as ke:
```

Comment:
As in INGK-463: This try-except changes the function behavior. The parent function can't return None, but this function does.

Reply:
It's solved here: [INGK-464](https://github.com/ingeniamc/ingenialink-python/pull/213#discussion_r996689087)